streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -271,7 +282,10 @@ object CheckpointReader extends Logging {
     })
 
     // If none of checkpoint files could be read, then throw exception
-    throw new SparkException("Failed to read checkpoint from directory " + checkpointPath)
+    if (!ignoreReadError) {
+      throw new SparkException("Failed to read checkpoint from directory " + checkpointPath)
```

Comment:
Do you think that we should log a warning message in the case where we ignore the error?